networkx/algorithms/bipartite/faster_maximum_prioirity_matching.py

```diff
@@ -0,0 +1,532 @@
+    >>> G.add_nodes_from(nodes)
+    >>> G.add_edges_from(edges)
+    >>> nx.set_node_attributes(G, nodes_attrs)
+    >>> nx.set_edge_attributes(G, edges_attrs)
+    >>> matching_edges = nx.get_edge_attributes(G, 'isMatched')
+    >>> matching_info = nx.get_edge_attributes(G, 'isMatched')
+    >>> reverse_path(G , ['8','9','10','12','11','7','6','3','2','1'])
+    >>> matching_edges = nx.get_edge_attributes(G, "isMatched")
+    >>> print(matching_edges[('8', '9')])
+    True
+    >>> print(matching_edges[('9', '10')])
+    False
+    >>> print(matching_edges[('10', '12')])
+    True
+    >>> print(matching_edges[('11', '12')])
+    False
+    >>> print(matching_edges[('7', '11')])
+    True
+    >>> print(matching_edges[('6', '7')])
+    False
+    >>> print(matching_edges[('3', '6')])
+    True
+    >>> print(matching_edges[('2', '3')])
+    False
+    >>> print(matching_edges[('1', '2')])
+    True
+
+
+
+    """
+
+    matching_nodes = nx.get_node_attributes(G, "isMatched")
+    matching_edges = nx.get_edge_attributes(G, "isMatched")
+    for i in range(0, len(path) - 1):
+        if (path[i], path[i + 1]) in matching_edges:
+            nx.set_edge_attributes(
+                G,
+                {
+                    (path[i], path[i + 1]): {
+                        "isMatched": not matching_edges[(path[i], path[i + 1])]
+                    }
+                },
+            )
+        else:
+            nx.set_edge_attributes(
+                G,
+                {
+                    (path[i + 1], path[i]): {
+                        "isMatched": not matching_edges[(path[i + 1], path[i])]
+                    }
+                },
+            )
+
+    nx.set_node_attributes(G, {path[0]: {"isMatched": not matching_nodes[path[0]]}})
+    nx.set_node_attributes(G, {path[-1]: {"isMatched": not matching_nodes[path[-1]]}})
+
+
+def find_maximum_priority_matching_bipartite(G: nx.Graph):
+    """
+    "Faster Maximium Priority Matchings in Bipartite Graphs" by Tarjan, Robert E.
+
+    Programmers: Roi Meshulam and Liroy Melamed
+
+    Our find_maximum_priority_matching_bipartite gets bipartite graph and returns the maximum priority matching
+
+    :param G: nx.Graph
+    :return: A list of edges
+
+    Tests:
+    >>> G = nx.Graph()
+    >>> nodes = ['1', '2', '3', '4', '5', '6']
+    >>> edges = [('1', '2'), ('1', '4'), ('3', '6'),('5', '6')]
+    >>> nodes_attrs = {'1': {"priority": 1, "Group": 1},'2': {"priority": 3, "Group": 2},'3': {"priority": 2, "Group": 1},'4': {"priority": 4, "Group": 2},'5': {"priority": 5, "Group": 1},'6': {"priority": 6, "Group": 2}}
```

Comment:
The attibute "Group" is not relevant  - it should not be a part of the input. 
Please look at other functions dealing with bipartite graphs in  networkx, and see how they define bipartite graphs.